heat/core/operations.py

```diff
@@ -509,6 +515,494 @@ def sum(x, axis=None, out=None):
 
 
+def merge_means(mu1, n1, mu2, n2):
+    """
+    Function to merge two means by pairwise update
+
+    Parameters
+    ----------
+    mu1 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n1 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu1
+    mu2 : 1D ht.tensor or 1D torch.tensor
+          Calculated mean
+    n2 : 1D ht.tensor or 1D torch.tensor
+         number of elements used to calculate mu2
+
+    Returns
+    -------
+    mean of combined set
+    number of elements in the combined set
+
+    References
+    ----------
+    [1] J. Bennett, R. Grout, P. Pebay, D. Roe, D. Thompson, Numerically stable, single-pass, parallel statistics algorithms,
+        IEEE International Conference on Cluster Computing and Workshops, 2009, Oct 2009, New Orleans, LA, USA.
+    """
+    delta = mu2.item() - mu1.item()
+    n1 = n1.item()
+    n2 = n2.item()
+    return mu1 + n2 * (delta / (n1 + n2)), n1 + n2
+
+
+def mean(x, axis=None):
+    """
+    Calculates and returns the mean of a tensor.
+    If a axis is given, the mean will be taken in that direction.
+
+    Parameters
+    ----------
+    x : ht.tensor
+        Values for which the mean is calculated for
+    axis : None, Int, iterable
+           axis which the mean is taken in.
+           Default: None -> mean of all data calculated
+
+    Returns
+    -------
+    ht.tensor containing the mean/s, if split, then split in the same direction as x.
+    """
+
+    def reduce_means_elementwise(output_shape_i):
+        """
+        Function to combine the calculated means together.
+        This does an element-wise update of the calculated means to merge them together using the merge_means function.
+        This function operates using x from the mean function paramters
+
+        Parameters
+        ----------
+        output_shape_i : iterable
+                        iterable with the dimensions of the output of the mean function
+
+        Returns
+        -------
+        ht.tensor of the calculated means
+        """
+        if x.lshape[x.split] != 0:
+            mu = __local_operation(torch.mean, x, out=None, dim=axis)
+        else:
+            mu = tensor.zeros(output_shape_i)
+
+        mu_to_combine = tensor.zeros((x.comm.Get_size(),) + tuple(output_shape_i))
+        mu_tot = tensor.zeros((x.comm.Get_size(),) + tuple(output_shape_i))
+
+        n_for_merge = tensor.zeros(x.comm.Get_size())
+        n2 = tensor.zeros(x.comm.Get_size())
+        mu_to_combine[x.comm.Get_rank()] = mu
+        n2[x.comm.Get_rank()] = x.lshape[x.split]
+        x.comm.Allreduce(mu_to_combine, mu_tot, MPI.SUM)
+        x.comm.Allreduce(n2, n_for_merge, MPI.SUM)
+
+        sz = x.comm.Get_size()
+        rem1 = rem2 = 0
+        while True:
+            if sz % 2 != 0:
+                if rem1 != 0 and rem2 == 0:
+                    rem2 = sz - 1
+                elif rem1 == 0:
+                    rem1 = sz - 1
+            splt = int(sz / 2)
```

Comment:
you can use // here, does integer division but more efficient due to low-level implementation